### ISSUE TITLE
MQTTv5: Mqttv5PahoMessageDrivenChannelAdapter unsubscribes from topics even when cleanStart/cleanSession is false

### ISSUE THREAD

**Issue by mths1** [spring-projects/spring-integration#3955]
**In what version(s) of Spring Integration are you seeing this issue?**

sprint-boot 2.7.5
spring-integration-mqtt 5.5.15
org.eclipse.paho.mqttv5.client 1.2.5

**Describe the bug**
Mqttv5PahoMessageDrivenChannelAdapter unsubscribes from all topics, even if cleanStart/cleanSession is set to false, thus not receiving offline messages after restart. Similar behaviour to https://jira.spring.io/browse/INT-3900 but for MQTT v5.

**To Reproduce**
- Create a Mqttv5PahoMessageDrivenChannelAdapter with setCleanStart(false) and setSessionExpiryInterval with e.g. 5000secs
- Subscribe to some topics
- Stop application
- On the broker you will see that it unsubscribes from all topics

**Expected behavior**

The Mqttv5PahoMessageDrivenChannelAdapter does not unsubscribe in inStop.
Mqttv5PahoMessageDrivenChannelAdapter.java:

```
protected void doStop() {
		this.topicLock.lock();
		String[] topics = getTopic();
		try {
			if (this.mqttClient != null && this.mqttClient.isConnected()) {
				this.mqttClient.unsubscribe(topics).waitForCompletion(getCompletionTimeout());

				if (getClientManager() == null) {
					this.mqttClient.disconnect().waitForCompletion(getCompletionTimeout());
				}
			}
		}
		catch (MqttException ex) {
			logger.error(ex, () -> "Error unsubscribing from " + Arrays.toString(topics));
		}
		finally {
			this.topicLock.unlock();
		}
	}
```

MqttPahoMessageDrivenChannelAdapter (v3) does a check:
```
	protected synchronized void doStop() {
		try {
			if (this.consumerStopAction.equals(ConsumerStopAction.UNSUBSCRIBE_ALWAYS)
					|| (this.consumerStopAction.equals(ConsumerStopAction.UNSUBSCRIBE_CLEAN)
					&& this.cleanSession)) {

				this.client.unsubscribe(getTopic());
			}
		}
```

Proposed fix:
```
    @Override
    protected void doStop() {
        this.topicLock.lock();
        String[] topics = getTopic();
        try {
            if (this.mqttClient != null && this.mqttClient.isConnected()) {
                /** FIX */
                if (connectionOptions.isCleanStart()) {
                    this.mqttClient.unsubscribe(topics).waitForCompletion(getCompletionTimeout());
                }
                /** FIX */
                this.mqttClient.disconnect().waitForCompletion(getCompletionTimeout());
            }
        }
        catch (MqttException ex) {
            logger.error(ex, () -> "Error unsubscribing from " + Arrays.toString(topics));
        }
        finally {
            this.topicLock.unlock();
        }
    }
```

**Comment by artembilan** [comment]
Confirmed.
Please, consider to contribute the fix: https://github.com/spring-projects/spring-integration/blob/main/CONTRIBUTING.adoc

**Comment by iamceph** [comment]
After trying this out, we figured that after the client reconnects (for example, when the network is cut-off), the subscribed channels are no longer receiving messages. Is there something we are missing or we should do further investigation? 

Thanks you, @mths1 @artembilan

**Comment by artembilan** [comment]
@iamceph ,

It might be better to start from StackOveflow question with much more info about your use-case and configuration then trying to connect your problem with something what is marked as fixed over here.

**Comment by iamceph** [comment]
@artembilan Just did that, thanks. :)

**Comment by whizyrel** [comment]
Good day everyone was this fixed?
I am still experiencing the issue

**Comment by artembilan** [comment]
@whizyrel ,

as you see the issue is fixed and that was in `6.0.1` version.
Please, make sure that you use the latest Spring Integration.

Otherwise, please, raise a new GH issue with much more info to determine what is going with current state of code in the framework.

**Comment by whizyrel** [comment]
I have 6.4.1 and it is no different @artembilan. It is no different for 6.0.1...

```gradle
...
	implementation 'org.springframework.integration:spring-integration-mqtt:6.4.1'
...
```

Perhaps, I am doing something wrong
